scripts/quests/bastok/Return_to_the_Depths.lua

```diff
@@ -0,0 +1,256 @@
+                    if npcUtil.tradeHasExactly(trade, xi.items.BULB_OF_MISAREAUX_GARLIC) and
+                    quest:getVar(player, 'prog') == 1 then
+                        return quest:progressEvent(99, 1, 1, 2, 0, 1661, 6396174, 4095, 4)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'prog') == 2 then
+                        return quest:progressEvent(46)
+                    elseif quest:getVar(player, 'prog') == 4 then
+                        return quest:progressEvent(100)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [99] = function(player, csid, option, npc)
+                    print("Progress (Finish)!")
+                    player:confirmTrade()
+                    npcUtil.giveKeyItem(player, xi.ki.LETTER_FROM_MUCKVIX)
+                    npcUtil.giveCurrency(player, "gil", 2000)
+                    quest:setVar(player, 'prog', 2)
+                end,
+
+                [100] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.PROVIDENCE_POT)
+                    player:delKeyItem(xi.ki.LETTER_FROM_MAGRIFFON)
+                    player:delKeyItem(xi.ki.LETTER_FROM_MUCKVIX)
+                    player:messageSpecial(zones[player:getZoneID()].text.KEYITEM_LOST, xi.ki.PROVIDENCE_POT)
+                    player:messageSpecial(zones[player:getZoneID()].text.KEYITEM_LOST, xi.ki.LETTER_FROM_MAGRIFFON)
+                    player:messageSpecial(zones[player:getZoneID()].text.KEYITEM_LOST, xi.ki.LETTER_FROM_MUCKVIX)
+                    npcUtil.giveKeyItem(player, xi.ki.PUNGENT_PROVIDENCE_POT)
+                    quest:setVar(player, 'prog', 5)
+                end,
+            },
+        },
+
+        [xi.zone.KAZHAM] =
+        {
+            ['Magriffon'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'prog') == 2 then
+                        return quest:progressEvent(299, 66, 630, 10000, 0, 65994751, 7080303, 4095, 4)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { {"gil", 10000} } ) and
+                    quest:getVar(player, 'prog') == 3 then
+                        return quest:progressEvent(301, 75, 630, 10000, 0, 65994751, 7080303, 4095, 4)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [299] = function(player, csid, option, npc)
+                    return quest:setVar(player, 'prog', 3)
+                end,
+
+                [301] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    npcUtil.giveKeyItem(player, {xi.ki.PROVIDENCE_POT, xi.ki.LETTER_FROM_MAGRIFFON})
```

Comment:
Might need to add spaces around these two KI with the new table formating sanity checks? 
I could be shooting in the dark here.

Reply:
Yes -- Fixed with other sanity checks